library/billing/src/HCFA_1500.php

```diff
@@ -0,0 +1,678 @@
+ * @copyright Copyright (c) 2011 Rod Roark <rod@sunsetsystems.com>
+ * @copyright Copyright (C) 2018 Stephen Waite <stephen.waite@cmsvt.com>
+ * @link http://www.open-emr.org
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Billing;
+
+require_once(dirname(__FILE__) . "/../../invoice_summary.inc.php");
+
+use OpenEMR\Billing\Claim;
+
+class HCFA_1500
+{
+    private $hcfa_curr_line = 1;
+    private $hcfa_curr_col = 1;
+    private $hcfa_data = '';
+    private $hcfa_proc_index = 0;
+
+    /**
+     * take the data element and place it at the correct coordinates on the page
+     *
+     * @global int $hcfa_curr_line
+     * @global int $hcfa_curr_col
+     * @global type $hcfa_data
+     * @param type $line
+     * @param type $col
+     * @param type $maxlen
+     * @param type $data
+     * @param type $strip   regular expression for what to strip from the data. period and has are the defaults
+     *                      02/12 version needs to include periods in the diagnoses hence the need to override
+     */
+    public static function put_hcfa($line, $col, $maxlen, $data, $strip = '/[.#]/')
+    {
+        global $hcfa_curr_line, $hcfa_curr_col, $hcfa_data;
+        if ($line < $hcfa_curr_line) {
+            die("Data item at ($line, $col) precedes current line.");
+        }
+
+        while ($hcfa_curr_line < $line) {
+            $hcfa_data .= "\n";
+            ++$hcfa_curr_line;
+            $hcfa_curr_col = 1;
+        }
+
+        if ($col < $hcfa_curr_col) {
+            die("Data item at ($line, $col) precedes current column.");
+        }
+
+        while ($hcfa_curr_col < $col) {
+            $hcfa_data .= " ";
+            ++$hcfa_curr_col;
+        }
+
+        $data = preg_replace($strip, '', strtoupper($data));
+        $len = min(strlen($data), $maxlen);
+        $hcfa_data .= substr($data, 0, $len);
+        $hcfa_curr_col += $len;
+    }
+
+
+    public static function gen_hcfa_1500($pid, $encounter, &$log)
+    {
+        global $hcfa_data, $hcfa_proc_index;
+
+        $hcfa_data = '';
+        $hcfa_proc_index = 0;
+
+        $today = time();
+        $claim = new Claim($pid, $encounter);
+
+        $log .= "Generating HCFA claim $pid-$encounter for " .
+            $claim->patientFirstName() . ' ' .
+            $claim->patientMiddleName() . ' ' .
+            $claim->patientLastName() . ' on ' .
+            date('Y-m-d H:i', $today) . ".\n";
+
+        while ($hcfa_proc_index < $claim->procCount()) {
+            if ($hcfa_proc_index) {
+                $hcfa_data .= "\014"; // append form feed for new page
+            }
+
+            HCFA_1500::gen_hcfa_1500_page($pid, $encounter, $log, $claim);
+        }
+
+        $log .= "\n";
+        return $hcfa_data;
+    }
+
+    private function gen_hcfa_1500_page($pid, $encounter, &$log, &$claim)
+    {
+        global $hcfa_curr_line, $hcfa_curr_col, $hcfa_data, $hcfa_proc_index;
```

Comment:
Need to remove these globals and prepend them below with $this-> when use them:
$this->$hcfa_curr_line
$this->$hcfa_curr_col
$this->$hcfa_data
$this->$hcfa_proc_index